### ISSUE TITLE
Tuning direction is never reconciled with metric greater_is_better

### ISSUE THREAD

**Issue by nbx-liz** [nbx-liz/LizyML#258]
## Summary

`tuning.optuna.params.direction` defaults to `"minimize"` and is never reconciled with the objective metric's `greater_is_better`. A config that selects a higher-is-better metric is accepted with no error and no warning, and Optuna then optimises in the wrong direction.

## Evidence (HEAD `5712f41`)

- `lizyml/config/schema.py:498` — `direction: Literal["minimize", "maximize"] = "minimize"`
- `lizyml/core/_model_tuning.py:207` — the objective metric is `evaluation.metrics[0]` via `parse_metric_entry`
- `lizyml/core/_model_tuning.py:456` — `score: float = metrics["raw"]["oof"][metric_name]` is returned as the objective value, unsigned
- `lizyml/core/_model_tuning.py:495` -> `lizyml/tuning/tuner.py:142` — `create_study(direction=self.direction)`
- `grep -rn "greater_is_better" lizyml` finds no reconciliation with `direction` anywhere in the codebase. `BaseMetric.greater_is_better` (`lizyml/metrics/base.py:36`) is declared by every metric but is not consumed on the tuning path.

## Reproduction (executed against the shipped code)

```python
cfg = LizyMLConfig.model_validate({
    "config_version": 1,
    "task": "binary",
    "data": {"target": "y"},
    "split": {"method": "kfold", "n_splits": 3},
    "model": {"name": "lgbm"},
    "evaluation": {"metrics": ["auc"]},
    "tuning": {"optuna": {"params": {"n_trials": 5}}},
})
cfg.tuning.optuna.params.direction   # -> "minimize"
get_metric("auc").greater_is_better  # -> True
```

Validation succeeds with no error and no warning.

## Impact

With `evaluation.metrics: ["auc"]` (or `accuracy`, `f1`, `r2`, `precision_at_k`, ...) and `direction` left at its default, `Model.tune()` minimises the metric. `best_params` / `best_score` then describe the worst configuration found, and the result is indistinguishable from a successful run: no exception, no warning, and a plausible-looking `TuningResult`.

This is a silent-pass defect: the failure is invisible in the artifacts the user inspects.

## Suggested direction

Two parts, in order of value:

1. Derive `direction` from `Metric.greater_is_better` when the user did not set it explicitly.
2. When the user did set it explicitly and it contradicts the metric, reject with `ErrorCode.CONFIG_INVALID` rather than proceeding.

Both parts change public Config semantics, so this needs a HISTORY.md Proposal before implementation (CLAUDE.md section 2, Change Gate). The Proposal should cover backward compatibility for configs that currently pass an explicit `direction` matching the old behaviour.

## Relevant SKILL

`skills/tuning/SKILL.md`, `skills/metrics/SKILL.md`, `skills/config-and-specs/SKILL.md`

## DoD

- Objective metric and study direction cannot disagree at runtime.
- A regression test pins the reported behaviour: a config with a higher-is-better metric and no explicit `direction` must not produce a minimising study.
- A test covers the explicit-and-contradictory case raising `CONFIG_INVALID`.

**Comment by nbx-liz** [comment]
### Measured scope: 10 of 22 task-metric entries are wrong at the default direction

The body names `auc` and gives `accuracy, f1, r2, precision_at_k, ...` as
examples. The population is now closed and executed, so the "..." can be
replaced by an exhaustive list.

Population: every `(task, metric)` pair in `_TASK_METRICS` crossed with both
values of `direction` — **22 entries x 2 directions = 44 cells, all 44
classified, 0 unclassified.**

The guard search decides whether a conflicting cell is a defect or merely
rejected: **0 sites in `lizyml/` name both `direction` and `greater_is_better`
within a 6-line window**, so every conflicting pair is unguarded.

| verdict | cells |
|---|---|
| `AGREES` | 22 |
| `UNGUARDED-DEFECT` | 22 |
| `GUARDED` | 0 |
| `CANNOT-TELL` | 0 |

Of those, the ones that bite at the **default** `direction: "minimize"` — i.e.
what a user gets without writing `direction` at all:

**10 of 22 entries.**

- `regression`: `r2`
- `binary`: `accuracy`, `auc`, `auc_pr`, `f1`, `precision_at_k`
- `multiclass`: `accuracy`, `auc`, `auc_pr`, `f1`

The remaining 12 are lower-is-better metrics, which agree with the default by
coincidence rather than by any reconciliation.

### Reproduction

```python
from lizyml.config.schema import OptunaParamsConfig
from lizyml.metrics.registry import _TASK_METRICS, get_metric

default_dir = OptunaParamsConfig.model_fields["direction"].default   # "minimize"
bad = [(t, n) for t, ms in _TASK_METRICS.items() for n in sorted(ms)
       if get_metric(n).greater_is_better != (default_dir == "maximize")]
print(len(bad), sum(len(ms) for ms in _TASK_METRICS.values()))   # -> 10 22
for t, n in bad:
    print(t, n)
```

Positive control: `(binary, auc, minimize)` — the case this issue was filed on —
is re-detected by the same pass, so a run that found nothing would have been
visibly broken rather than silently clean.

### Consequence for the DoD

The DoD's regression test should be parametrised over the 10 entries above, not
written against `auc` alone. Deriving `direction` from `greater_is_better`
(suggestion 1 in the body) fixes all 10 at once; a fix validated only on `auc`
would leave the other 9 unpinned.

**Comment by nbx-liz** [comment]
### The 10-of-22 figure is now executed, not inferred

My earlier comment gave the scope as a measurement. It was not: it read each
metric's `greater_is_better`, compared it with the default `direction`, and
searched for a reconciling validator. That predicts what a tuning run would do;
it does not observe one.

All 44 `(task, metric, direction)` cells have now been run as real
`Model.tune()` jobs — 6 trials each, a search space over `learning_rate` and
`num_leaves` wide enough that trials differ in quality — and the question asked
of each is: **is the trial Optuna selected the one this metric would call best?**

```
44 cells, all classified, 0 CANNOT-TELL
  22  AGREES            selected the extreme the metric prefers
  22  UNGUARDED-DEFECT  selected the opposite extreme
```

The cell this issue was filed on, observed rather than predicted:

```
(binary, auc, minimize)
  trial scores : [0.4704, 0.4802, 0.5170, 0.5339, 0.5372, 0.5372]
  best_score   : 0.4704      <- the lowest AUC of the six
```

**Cells that select the wrong trial at the default direction — what a user gets
without writing `direction` at all — remain 10 of 22, now by observation:**

`regression/r2` · `binary/{accuracy, auc, auc_pr, f1, precision_at_k}` ·
`multiclass/{accuracy, auc, auc_pr, f1}`

The static pass and the executed pass agree on the figure. That agreement is
what makes it worth reporting; the earlier comment claimed it before it was
earned.

Reproduced by `r1_d3b_execute.py` in the discovery working set, which reads
`_TASK_METRICS` and `OptunaParamsConfig` from the shipped code and writes
nothing to the repository.

**Comment by nbx-liz** [comment]
Completed by H-0099, following the merged PR titled `fix(tuning): reconcile direction and reject inert dimensions` (reviewed candidate `8b739ce7cd6a4ce66662cd92072d293c4ed646d5`).

Omitted/null direction now derives from the first effective metric, including parameterized metrics and Config round trips. Explicit contradictions and incompatible supplied/persisted studies raise CONFIG_INVALID before study execution. Registry-derived real tune tests cover all 22 task/metric pairs and the correct selected extremum. Candidate full suite: 7830 passed, 230 skipped; independent local review: APPROVE.
